lib/config/secret.js

```diff
@@ -0,0 +1,28 @@
+module.exports = getSecret
+
+var internals = getSecret.internals = {}
+internals.randomstring = require('randomstring')
+
+function getSecret (state, callback) {
+  state.db.config.get()
+
+  .then(function (config) {
+    if (config.secret) {
+      state.secret = config.secret
+      return callback()
+    }
+
+    config.secret = internals.randomstring.generate({
+      charset: 'hex'
+    })
+
+    state.db.config.set(config)
+
+    .then(function () {
```

Comment:
similar to the other nested `.then` case you commented, the `callback` would be called twice if we would remove the nesting. Besides the `config` variable would not be accessible, we would need to define it outside. 

We could workaround this by moving the "set secret" logic into a utils function, but I think this is simple enough asis